pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1001,6 +1003,98 @@ func setMetricAnyValue(item *metrics.MetricItem, val interface{}) {
 	}
 }
 
+// Run a periodic post of health checks to the controller
+func hardwareHealthTimerTask(ctx *zedagentContext, handleChannel chan interface{}) {
+	iteration := 0
+	log.Functionln("starting report health check timer task")
+	publishΗealthChecksReport(ctx, iteration)
+
+	interval := time.Duration(ctx.globalConfig.GlobalValueInt(types.HardwareHealthInterval)) * time.Second
+	max := float64(interval)
+	min := max * 0.3
+	ticker := flextimer.NewRangeTicker(time.Duration(min), time.Duration(max))
+	// Return handle to caller
+	handleChannel <- ticker
+
+	wdName := agentName + "hardwarehealth"
+
+	// Run a periodic timer so we always update StillRunning
+	stillRunning := time.NewTicker(600 * time.Second)
```

Comment:
The watchdog on the touch files is 5 minutes and here you have a 10 minute timer. That doesn't work. Elsewhere do a 15 or 30 second timer for stillRunning.

Reply:
I copied what we have for publishing metrics.